cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -468,6 +497,23 @@ struct index : cuvs::neighbors::index {
  *   auto distances = raft::make_device_matrix<float>(res, n_queries, k);
  *   cagra::search(res, search_params, index, queries, neighbors, distances);
  * @endcode
+ *   In the above example, the dataset is attached to the index and the index is ready for
```

Comment:
I think it is sufficient to keep the example in the docstring of `attach_dataset_on_build`.